conda_smithy/configure_feedstock.py

```diff
@@ -588,6 +589,17 @@ def _render_ci_provider(
             if ver:
                 os.environ["DEFAULT_LINUX_VERSION"] = ver
 
+        # detect if `compiler('cuda')` is used in meta.yaml,
+        # and set appropriate environment variable
+        with open(
+            os.path.join(forge_dir, forge_config["recipe_dir"], "meta.yaml")
+        ) as f:
+            meta_lines = f.readlines()
+        for ml in meta_lines:
+            # looking for `compiler('cuda')` with both quote variants
+            if re.match(r"compiler\((\"cuda\"|\'cuda\')\)", ml):
+                os.environ["CF_CUDA_ENABLED"] = "True"
```

Comment:
regex pattern could be compiled, but didn't think that's (performance-)relevant here.

Reply:
let's compile it. it might not be relevant today, but I don't want to leave it for another to trip over it some day.

Reply:
Compiling a regular expression only makes sense if it's reused (often) - as it would have been for the individual lines. Since we now have `meta.yaml` as one string, compiling makes no difference anymore.